### ISSUE TITLE
[BUG] Demonic Pact stacks with Totem of Wrath and Flametongue Totem

### ISSUE THREAD

**Issue by Zephancode** [Zephancode/wotlk-warlock#30]
## 1. Describe the bug
Demonic Pact stacks with Totem of Wrath and Flametongue Totem

### 1.1 To Reproduce
1. Join a Party with a Shaman
2. Open character pane and note spell power
3. Pet attacks -> Crit: New spell power
4. Shaman drops totems -> Does sp go up?

### 1.2 Source Material
- https://github.com/Zephancode/wotlk-warlock/issues/26#issuecomment-1165842257

### 1.3 Game Version:

24/06/2022

### 2.1 Source expected behavior
- See the following youtube video of a 3.3.3a lich king kill for evidence of DP overwriting ToW (notably though we can't see SP, but the buff does disappear): https://www.youtube.com/watch?v=Dzp83WnN9yA
- The EJ demonology guide mentions it not stacking in section 2,1 Demonic Pact: Basics http://web.archive.org/web/20100807225539/http://elitistjerks.com/f80/t100099-demonology_releasing_demon_you/
- Patch 3.0.2 undocumented changes: "Spell power increase does not stack with Focus Magic (sic), Improved Divine Spirit, Flametongue Totem, or Demonic Pact (see discussion)" https://gist.github.com/BradBlondebeard/51de640accc78e63e87a4e6d0ee73783

### 2.2 Additional context
NA

**Comment by nodddle** [comment]
with neither DP nor ToW I have 626 SP, my corruption ticks for 318.
with DP and no ToW I have 694 SP and my corruption ticks for 334.
with ToW and no DP I have 766 SP and my corruption ticks for 350.
with ToW AND DP I have 850 SP and my corruption ticks for 370.
I put up DP, I have 694 SP and a corruption that hits for 334. if I then drop ToW I have 834 SP and a corruption that ticks for 366.

looks like DP/ToW is stacking and not just snapshotting the DP and then being overwritten by it like was previously thought?

level 70 ToW = 140 SP
my SP with 626 SP + DP = 694
my SP with ToW = 766
my SP with ToW + DP = 850

this means that ToW is a stronger buff than my DP and it's possible that this is the reason why they are stacking. To test this I will buff myself with fel armor (186SP),  blood fury (143SP), and glyph of lifetap(28SP), as well as use rank 1 ToW which is 100 SP instead of 140.

with neither DP nor ToW I have 983 SP, my corruption ticks for 400.
with DP and no ToW I have 1091 SP and my corruption ticks for 426.
1091 - 983 = 108 which means that my DP buff is greater than a ToW at 100 SP.
with ToW then DP I have 1202 SP and my corruption ticks for 451.

tldr yes, I think DP not only stacks with ToW but contributes to DP spell power as well.
